lib/model/settings.dart

```diff
@@ -44,7 +45,39 @@ enum BrowserPreference {
   external,
 }
 
-extension GlobalSettingsHelpers on GlobalSettingsData {
+/// Store for the user's account-independent settings.
+///
+/// From UI code, use [GlobalStoreWidget.settingsOf] to get hold of
+/// an appropriate instance of this class.
+class GlobalSettingsStore extends ChangeNotifier {
+  GlobalSettingsStore({
+    required GlobalStoreBackend backend,
+    required GlobalSettingsData data,
+  }) : _backend = backend, _data = data;
+
+  final GlobalStoreBackend _backend;
+
+  /// A cache of the [GlobalSettingsData] singleton in the underlying data store.
+  GlobalSettingsData _data;
+
+  /// The user's choice of [ThemeSetting];
+  /// null means the device-level choice of theme.
+  ///
+  /// See also [setThemeSetting].
+  ThemeSetting? get themeSetting => _data.themeSetting;
+
+  /// The user's choice of [BrowserPreference];
+  /// null means use our default choice.
+  ///
+  /// Consider using [effectiveBrowserPreference] or [getUrlLaunchMode].
+  ///
+  /// See also [setBrowserPreference].
+  BrowserPreference? get browserPreference => _data.browserPreference;
```

Comment:
A reason that `effectiveBrowserPreference` is separate from this was that we didn't opt into implementing a custom `GlobalSettingsData`.

Now that `GlobalSettingsData _data` is nicely hidden as a part of the implementation detail, we can rename `effectiveBrowserPreference` to `browserPreference`, replacing it, and skip the part of the dartdoc redirecting to `effectiveBrowserPreference`.

Reply:
I guess keeping `browserPreference` nullable is consistent with `setBrowserPreference`, but I think we don't need to expose a way to set it to `null`.

Reply:
Yeah, I think hiding the raw `browserPreference` behind (a renamed) `effectiveBrowserPreference` could be a good follow-up change. Then  `setBrowserPreference` would narrow its parameter type to match.

I'll leave that out of this PR, though, because I think it's not on the path to #1409.